### PR TITLE
openthread: adopted Kconfig to Zephyr OpenThread choice configuration

### PR DESCRIPTION
**Pull request by tejlmand**
This ensures that security selected symbols are not stuck when users
reconfigure the system.

Manifest updated to corresponding changes in nrfxlib and sdk-zephyr.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

### CODE REVIEW

**Review by pdunaj — APPROVED** [review]

**Comment by tejlmand** [comment]
CI-test-label removed as this only concerns Kconfig dependency changes.

File: subsys/net/openthread/Kconfig [hunk 1]
```diff
@@ -12,10 +12,21 @@ if NET_L2_OPENTHREAD
...
 menu "OpenThread"
 
+config NORDIC_SECURITY_PROMPTLESS
+	bool
+	default y if !CUSTOM_OPENTHREAD_SECURITY
+
+choice OPENTHREAD_SECURITY
+	depends on !OPENTHREAD_SECURITY_INTERNAL
+
+config OPENTHREAD_NRF_SECURITY_CHOICE
```
Comment (MarekPorwisz) [inline subsys/net/openthread/Kconfig:22]: If I am not mistaken also nrf/samples/openthread/common/overlay-ot-defaults.conf file needs to be updated. Currently it states:
# Disable vanilla mbedtls in favor of custom backends
CONFIG_OPENTHREAD_MBEDTLS=n

But I think that now it should select the OPENTHREAD_NRF_SECURITY_CHOICE.
All Thread nrf samples should use nRF security.

Comment (tejlmand) [reply subsys/net/openthread/Kconfig:22]: correct and thanks, extra commit pushed.

**Comment by nvlsianpu** [comment]
[DNM], included into #4527

**Review by Damian-Nordic — APPROVED** [review]

**Review by kkasperczyk-no — APPROVED** [review]

**Comment by nvlsianpu** [comment]
introduced within #4527

**Pull request closed without merging**